### PR TITLE
add manpage for aurpublish(1)

### PR DESCRIPTION
**Pull request by aminvakil**
Adds an asciidoc man page to the repository.

### CODE REVIEW

**Comment by aminvakil** [comment]
Basic functionality has been tested and can be verified in https://github.com/aminvakil/aurpublish/runs/1676164921?check_suite_focus=true.

**Comment by aminvakil** [comment]
I'm sure grammar, vocabulary and stuff can definitely be improved, right now it's basically a combination of `aurpublish -h` and [README.md](https://github.com/eli-schwartz/aurpublish/blob/master/README.md)

File: doc/aurpublish.1.asciidoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Name
+----
+aurpublish - PKGBUILD management framework for the Arch User Repository
+
+
+Synopsis
+--------
+'aurpublish' [options] PACKAGE
+'aurpublish' log [<options>] PACKAGE
+'aurpublish' setup
```
Comment (eli-schwartz) [inline doc/aurpublish.1.asciidoc:15]: These need to be separated by one empty line each.

File: doc/aurpublish.1.asciidoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+Description
+-----------
+Aurpublish is an utility which helps maintainers of aur packages.
```
Comment (eli-schwartz) [inline doc/aurpublish.1.asciidoc]: "... to submit updates" maybe?

File: doc/aurpublish.1.asciidoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+------------
+Commit PKGBUILDs in named subdirectories.
+Export them to the AUR with the aurpublish command, using the subtree push 
+stratagem. This preserves an independent history for third-party hosting, 
+pull requests... ;)
+
+Commands
+--------
+
+setup::
```
Comment (eli-schwartz) [inline doc/aurpublish.1.asciidoc]: Should be bolded (as with other options below too) by adding asterisks around it.
```suggestion
*setup*::
```

File: doc/aurpublish.1.asciidoc [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+-p <PACKAGE>::
+        Instead of publishing, pull changes from the AUR.
+        Can import packages into a new subtree.
+
+-s, --speedup <PACKAGE>::
+        Speedup future publishing by recording the subtree history during 
+        a push. This creates a merge commit and a second copy of all 
+        commits in the subtree. For more details, see the "--rejoin" 
+        option in git subtree.
```
Comment (eli-schwartz) [inline doc/aurpublish.1.asciidoc:50]: We can use the linkman macro from pacman to turn git-subtree into a manpage link... hmm.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]
Sorry for the lateness of the reply.

Some nitpicking below. Additionally, I'd like to remove the rendered manpage, and build it in the Makefile using `a2x`. The [community] package will provide it for most people.

File: doc/aurpublish.1.asciidoc (hunk 1 above)
Comment (aminvakil) [reply doc/aurpublish.1.asciidoc:15]: Fixed in https://github.com/eli-schwartz/aurpublish/pull/16/commits/8e17120aff596e68474c2e29b554e3dc3173120c.

File: doc/aurpublish.1.asciidoc (hunk 2 above)
Comment (aminvakil) [reply doc/aurpublish.1.asciidoc]: Fixed in https://github.com/eli-schwartz/aurpublish/pull/16/commits/03d4d0c704641ca9e9e703aa86dbad81ffb8da51.

File: doc/aurpublish.1.asciidoc (hunk 3 above)
Comment (aminvakil) [reply doc/aurpublish.1.asciidoc]: Fixed in https://github.com/eli-schwartz/aurpublish/pull/16/commits/1f859dabd05002408ee47a870c34bf9311d607b3.

Comment (aminvakil) [reply doc/aurpublish.1.asciidoc]: Fixed in https://github.com/eli-schwartz/aurpublish/pull/16/commits/1f859dabd05002408ee47a870c34bf9311d607b3.

Comment (aminvakil) [reply doc/aurpublish.1.asciidoc]: Fixed in https://github.com/eli-schwartz/aurpublish/pull/16/commits/1f859dabd05002408ee47a870c34bf9311d607b3.

File: doc/aurpublish.1.asciidoc (hunk 4 above)
Comment (aminvakil) [reply doc/aurpublish.1.asciidoc:50]: Sorry, but I'm not sure if I understood what you mean at all :)

**Comment by aminvakil** [comment]
> Sorry for the lateness of the reply.
> 
> Some nitpicking below. Additionally, I'd like to remove the rendered manpage, and build it in the Makefile using `a2x`. The [community] package will provide it for most people.

I've also removed rendered manpage in https://github.com/eli-schwartz/aurpublish/pull/16/commits/9016e6edcbf745bfb4cdc78f581ff1926ee23587.

File: doc/aurpublish.1.asciidoc (hunk 4 above)
Comment (eli-schwartz) [reply doc/aurpublish.1.asciidoc:50]: https://git.archlinux.org/pacman.git/tree/doc/asciidoc.conf

adding a2x --asciidoc-opts "-f doc/asciidoc.conf" will let you use macros defined in this attached file.

`linkman:git-subtree[1]` then turns into a nicely formatted reference.

See e.g. https://git.archlinux.org/arch-install-scripts.git/commit/?id=aeab95dbab2bac426ef8ab45056b83e2e6a888a0 for how I added the same to arch-install-scripts.

...

I could add this later, I suppose.

Comment (eli-schwartz) [reply doc/aurpublish.1.asciidoc:50]: See c126c24b92ea8e533c1953565ed978d2a9d14e24

(needs rebasing)

**Review by eli-schwartz — APPROVED** [review]
Looks good to me.

Can you finalize the PR by squashing this all down into one commit via `git rebase`? Then I'll merge this once I get home.

**Comment by aminvakil** [comment]
Done.

**Comment by eli-schwartz** [comment]
Thank you!

merged manually via cherry-pick.

**Comment by aminvakil** [comment]
@eli-schwartz Great, thanks for reviewing and of course merging!

**Pull request closed without merging**